getting-started/best-practices.md

```diff
@@ -0,0 +1,64 @@
+# "Best Practice" Guidelines
+
+## "Best Practice" Guidelines for Developers
+
+This document presents some guidelines that have been found useful. Please add your own suggestions!
+
+### Support Modules
+
+*   If your IOC application source directory contains '.c' files you should **definitely** consider moving them to a support module (either existing or new). A big problem in the past has been the proliferation of similar or identical '.c' files among multiple IOC applications. Help stamp out this practice now! To a lesser degree you should consider doing the same for any sequence programs in your application although these tend to be much more application-specific and thus impractical to move to a support module.
+*   Use `$EPICS_BASE/bin/makeBaseApp.pl -t support` to create a new support module. If you're writing a new message-based instrument support module `<path to asyn bin>/makeSupport.pl -t streamSCPI` is a good place to start.
+*   Nothing within the support 'modules' directory tree should be an 'App'. Nothing within the 'ioc' applications directory tree should be anything but an 'App'.
+
+### Device Support
+
+*   All message-based devices (with any expected lifetime) should be converted to use ASYN. If you are starting from existing devGpib support see [these guidelines for conversion](https://epics.anl.gov/modules/soft/asyn/R4-10/gpibCoreConversion/conversionNotes.html). If you are starting on brand-new support see [these notes](https://epics.anl.gov/modules/soft/asyn/R4-10/HowToDoSerial/tutorial.html) or have a look at the [stream device support](http://epics.web.psi.ch/software/streamdevice/).
+*   Support for a particular device should be in its own support module.
+
+### Database Files
+
+*   Support modules and IOC applications should install all .db files into their <top>/db directory. When an application uses a .db file from a support module that file may be copied from the support module `<top>/db` directory to the IOC application `<top>/db` directory. This is done by adding lines like `DB_INSTALLS += $(ASYN)/db/asynRecord.db` to the application `<top>/_appName_/Db/Makefile`. This allows all the .db files needed by an application to be found in a single location.
+*   Template files that are part of support modules do not need to be copied to your application Db source directory. The build system knows to look in support module db directories for template files.
+
+### Application Makefile
+
+*   Add support module .dbd files to your application using the Makefile `_xxxxxx_DBD += _yyyyy_.dbd` mechanism.
```

Comment:
Unfortunately this doesn't really resolve the issue: the problem is that the correct syntax in the Makefiles is
```makefile
DBDS += mydbd
mydbd_DBD += foo.dbd
```
so the name of the dbd file needs an underscore before the `DBD`; the issue before was that there was a _double_ underscore that was hard to see.. What I would propose then is either
```
xxxxx_DBD += yyyyy.dbd
```
or
```
<xxxxx>_DBD += <yyyyy>.dbd
```
or something like that.